python/ccf/ledger.py

```diff
@@ -222,16 +225,25 @@ def get_seqno(self) -> int:
         return self._version
 
 
-def _byte_read_safe(file, num_of_bytes):
+def _byte_read_safe(file, num_of_bytes=None):
```

Comment:
It may be cleaner to default to 0, in which case the branch is simpler. None is a useful default for mutable types, but for immutable types, it's not necessarily better than a good sentinel.